netlogo-gui/src/main/ide/JumpToDeclaration.scala

```diff
@@ -0,0 +1,91 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.ide
+
+import org.nlogo.core.{Femto, Token, TokenType, TokenizerInterface}
+import org.nlogo.editor.EditorArea
+
+
+object JumpToDeclaration {
+
+  def findTokenContainingPosition(source: String, position: Int): Option[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.find(p => p.start < position && p.end >= position)
+  }
+  def jumpToDeclaration(cursorPosition: Int, editorArea: EditorArea): Unit = {
+    val tokenOption = findTokenContainingPosition(editorArea.getText(), cursorPosition)
+    for(token <- tokenOption) {
+      val t = getDeclaration(token, editorArea.getText)
+      t.foreach(token => {
+        editorArea.select(token.start, token.end)
+      })
+    }
+  }
+
+  def getDeclaration(token: Token, source: String): Option[Token] = {
+    val tokens = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source).toSeq
+    val (globals, owns) = parseOwns(tokens)
+    val indexOfToken = tokens.indexWhere(t => t.start == token.start)
+    var stack = 0
+    for(i <- indexOfToken to 0 by -1) {
+      val tok = tokens(i)
+      if (tok.tpe.equals(TokenType.CloseBracket)) {
+        stack -= 1
+      }
+      if (tok.tpe.equals(TokenType.OpenBracket)) {
+        stack += 1
+      }
+      if(stack == 0 && tok.text.equalsIgnoreCase(token.text) && tokens(i - 1).text.equalsIgnoreCase("let")) {
```

Comment:
You gave an example yesterday where this worked (and it does work in many cases). In the following model, it works for the first use of `bar`, but not the second:

```NetLogo
to foo
  let bar 2
  show bar + 3
  ask turtles [
    fd bar
  ]
end
```